src/Type/Schema.php

```diff
@@ -341,6 +341,10 @@ private function loadType(string $typeName) : ?Type
 
         $type = $typeLoader($typeName);
 
+        if (is_callable($type)) {
```

Comment:
I think this should be a reasonable compromise (and also won't break for people having `__invoke` in their types - I've seen this weirdness somewhere):

```suggestion
        if (! $type instanceof Type && is_callable($type)) {
```

Reply:
Agree that we should not break `__invoke`-lovers.

If `! $type instanceof Type && ! is_callable($type)` would be invalid, right? Let's make sure we do the minimal amount of work required, as this part is somewhat performance-critical.

Reply:
I think `instanceof` check will be a happy path for 90+% of situations and `is_callable` won't even run for those cases. And `instanceof` is much cheaper than `is_callable`.

Also, those wishing to squeeze the last bit of performance can optimize their `typeLoaders` to always return instances of Type to hit the happy path.

Or am I missing something?

Reply:
> Also, people wishing to squeeze the last bit of performance can optimize their typeLoaders to always return instances of Type to hit the happy path.

Oh, that's an interesting point I hadn't considered--if we play all this right then it shouldn't matter what the client code returns from his type loader, as long as its a `Type` or a `callable`...

Reply:
I went with @vladar's suggestion. Looks pretty clean to me. Dunno why I didn't think of it myself, but I'm happy with it if you guys are. Please review again when you have a chance.